contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/writers/HDF5StringDataWriter.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.exec.store.hdf5.writers;
+
+import ch.systemsx.cisd.hdf5.IHDF5Reader;
+import org.apache.drill.common.types.TypeProtos;
+import org.apache.drill.exec.physical.resultSet.RowSetLoader;
+import org.apache.drill.exec.record.metadata.ColumnMetadata;
+import org.apache.drill.exec.record.metadata.MetadataUtils;
+import org.apache.drill.exec.store.hdf5.HDF5Utils;
+import org.apache.drill.exec.vector.accessor.ScalarWriter;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class HDF5StringDataWriter extends HDF5DataWriter {
+
+  private String[] data;
+
+  private ArrayList<String> listData;
+
+  private ScalarWriter rowWriter;
+
+  // This constructor is used when the data is a 1D column.  The column is inferred from the datapath
+  public HDF5StringDataWriter(IHDF5Reader reader, RowSetLoader columnWriter, String datapath) {
+    super(reader, columnWriter, datapath);
+    data = reader.readStringArray(datapath);
+    listData = new ArrayList<>();
+
+    listData.addAll(Arrays.asList(data));
+
+    fieldName = HDF5Utils.getNameFromPath(datapath);
+    ColumnMetadata colSchema = MetadataUtils.newScalar(fieldName, TypeProtos.MinorType.VARCHAR, TypeProtos.DataMode.OPTIONAL);
+    int index = columnWriter.addColumn(colSchema);
+    rowWriter = columnWriter.scalar(index);
+  }
```

Comment:
Not super clear why we read a row of data in the constructor. Or, are we reading all data for all rows? Given the implementation of `write()`, looks like the later.

If so, then how do we handle large files? 1G? 10G? We'd cache all the data in memory. If your data set consists of many such files, and Drill reads five of them per process, that could be quite a bit of cached data.

Is there a reason to not read the data incrementally row by row to minimize memory use?

Reply:
Unfortunately I didn't see any iterator classes in the HDF5 library.  It seemed like you had the choice of either:
1.   Reading the first element of a dataset
2.  Reading the entire dataset

There didn't seem to be any Iterator or obvious way of iterating over a dataset.